test/messagesIncoming.ts

```diff
@@ -80,13 +93,41 @@ describe('Incoming Messages', async () => {
     let messageBadData: Message;
     let messageEOA: Message;
     let messageEOANoAmount: Message;
-    let messageReentrant1: Message;
-    let messageReentrant2: Message;
 
     // Arrays of committed block headers and their IDs
     const blockHeaders: BlockHeader[] = [];
     const blockIds: string[] = [];
-    const blockSignatures: string[] = [];
+    let endOfEpochHeader: BlockHeader;
+    let endOfEpochHeaderLite: BlockHeaderLite;
+    let unflinalizedBlock: BlockHeader;
+    let prevBlockNodes: TreeNode[];
+
+    // Helper function to setup test data
+    function generateProof(message: Message, prevBlockDistance = 1): [string, BlockHeader, MerkleProof, MerkleProof] {
+        const messageBlockIndex = BLOCKS_PER_EPOCH - 1 - prevBlockDistance;
+        const messageBlockHeader = blockHeaders[messageBlockIndex];
+        const messageBlockLeafIndexKey = getLeafIndexKey(prevBlockNodes, blockIds[messageBlockIndex]);
+        const blockInHistoryProof = {
+            key: messageBlockLeafIndexKey,
+            proof: getProof(prevBlockNodes, messageBlockLeafIndexKey),
+        };
+        const messageID = computeMessageId(message);
+        const messageLeafIndexKey = getLeafIndexKey(messageNodes, messageID);
+        const messageInBlockProof = {
+            key: messageLeafIndexKey,
+            proof: getProof(messageNodes, messageLeafIndexKey),
+        };
+        return [messageID, messageBlockHeader, blockInHistoryProof, messageInBlockProof];
+    }
+    function generateProof2(message: Message): [string, MerkleProof] {
```

Comment:
It would be beneficial to distinguish the helpers here to make it clearer why one type of proof is being used. `generateProof<number>` doesn't convey much

Reply:
renamed to `generateJustMessageProof` although the two different flavors of proofs are getting combined in another update soon